File: Week 2 problemset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Get the text and assign it to the variable text as a string
    string text = get_string("Text: ");

    // Get the length of the string, avoiding to put it in the loop for opti
    int tlength = strlen(text);

    float letter = 0;
    float sentence = 0;
    float word = 1;
    for (int i = 0; i < tlength; ++i)
    {
        if (isalnum(text[i]))
        {
            ++letter;
        }
        if (isspace(text[i]))
        {
            ++word;
        }
        if ((isalpha(text[i - 1]) || (ispunct(text[i + 1]))) && (text[i] == '.' || text[i] == '!' || text[i] == '?'))
        {
            ++sentence;
        }
    }

    float l = (letter / word) * 100;
    float s = (sentence / word) * 100;

    float index = 0.0588 * l - 0.296 * s - 15.8;
    int rindex = round(index);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", rindex);
        //printf("Grade %f\n", index);
    }

    //printf("Letters: %f\n", letter);
    //printf("Sentences: %f\n", sentence);
    //printf("Words: %f\n", word);
}

File: Week 6 problemset 6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_text()
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    index = get_index(letters, words, sentences)
    check_grade(index)

# Get the text to check from the user
def get_text():
    while True:
        text = get_string("Text: ")
        if len(text) > 0:
            break
    return text

# Count each letters from the text if they're alpha numeric
def count_letters(text):
    letters = 0
    for c in text:
        if (c.isalnum()):
            letters += 1
    return letters

# Count words by counting each spaces, start from 1 to count the last word
def count_words(text):
    words = 1
    for c in text:
        if c == ' ':
            words += 1
    return words

# Count sentences by counting punctuations that end sentences
def count_sentences(text):
    sentences = 0
    sentence_end = {'.', '!', '?'}
    for c in text:
        if c in sentence_end:
            sentences += 1
    return sentences

# Use the data we got from other functions to calculate the Coleman-Liau index
def get_index(letters, words, sentences):
    l = (letters / words) * 100
    s = (sentences / words) * 100
    index = 0.0588 * l - 0.296 * s - 15.8
    return index


# Check in which grade the text belong to
def check_grade(index):
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")


main()